pht_federated/aggregator/services/discovery/adjust_tabular_dataset.py

```diff
@@ -0,0 +1,342 @@
+) -> DatasetStatistics:
+    """
+    Changes the column types and names of the local dataset according to the suggested changes provided by the
+    difference report while taking multiple different mismatching cases into account
+    :param local_dataset: Pandas dataframe of the tabular local dataset
+    :param local_dataset_stat: summary statistics of the given local dataset
+    :param difference_report: Lists differences between local and aggregated datasets and provides hints to resolve them
+    :return adjusted_dataset -> Dataset with applied column name changes
+    """
+
+    local_dataset_stat = local_dataset_stat.dict()
+
+    type_errors = [
+        column["hint"]
+        for column in difference_report["errors"]
+        if column["error"]["type"] == "type"
+    ]
+    type_diffs = [re.findall('"([^"]*)"', errors) for errors in type_errors]
+
+    name_errors = [
+        column["hint"]
+        for column in difference_report["errors"]
+        if column["error"]["type"] == "added"
+    ]
+    name_diffs = [re.findall('"([^"]*)"', errors) for errors in name_errors]
+    col_error_list = [error for error in name_diffs if len(error) > 1]
+
+    row_errors_list = []
+
+    for differences in type_diffs:
+        row_errors_list.append(
+            find_row_errors(local_dataset, differences[0], differences[1])
+        )
+
+    error_report = create_error_report(row_errors_list, col_error_list, "test_dataset")
+
+    print("Error Report : {}".format(error_report))
+
+
+def find_row_errors(local_dataset: pd.DataFrame, column_name: str, column_type: str):
+    """
+    Checks the column of the local_dataset where the error was found and searches for specific entries that
+    violate the desired/likely column type
+    :param local_dataset: Local dataset of the user that starts a proposal request over multiple stations
+    :param column_name: Name of the column where the type error occured
+    :param column_type: Suggested correct column type
+    :return: List[Tuple[error_index, error_value, column_name , column_type]]
+    """
+    row_errors_list = []
+
+    if column_type == "categorical":
+        row_errors_list = find_categorical_mismatch(
+            local_dataset, column_name, column_type
+        )
+    if column_type == "numeric":
+        row_errors_list = find_numerical_mismatch(
+            local_dataset, column_name, column_type
+        )
+    if column_type == "equal":
+        row_errors_list = find_equal_mismatch(local_dataset, column_name, column_type)
+    if column_type == "unique":
+        row_errors_list = find_unique_mismatch(local_dataset, column_name, column_type)
+    if column_type == "unstructured":
+        row_errors_list = find_unstructured_mismatch(
+            local_dataset, column_name, column_type
+        )
+
+    return row_errors_list
+
+
+def create_error_report(
+    row_errors_list: List[List[Tuple[str, str, str, str]]],
+    col_errors_list: List[List[str]],
```

Comment:
create specific objects for this inport -> i.e. HarmonizationError or ColumnHarmonizationResult etc.